The-Politico/artisan
# Actions: Archive, Backup, Delete, Downloads, GetProjectsArchive, Publish

**Actions**

- Open in Finder
- Download project
  - Sets `isUploaded: true`
- Delete Project
- Get Projects list (from Archive)
- Backup .ai files to S3
- Archive project (Backup + Delete)
- Publish project
  - Backup -> Output Share action -> Uploads each illustration's `ai2html-output` directory.

**Changes**

- Conditional updating of app settings
- `addIllustration` returns a promise
- Project operations are now in their own store file
- You can now pass optional configurations to a project with `/store/addProject()`

## How to test

Preflight: Ensure your publishing keys are in the `.local` env file and the staging bucket is in `.env`. Template files provided.

- `$ mkdir -p ~/Artisan/Projects`
- Click "Reset Store"
- Click "Set up settings"
- Sets the `projectFolder` param in the settings
  * Click console logging buttons as desired
  * Click "Download project"
- Project appears on your computer
  * Click "Open in Finder"
- Opens the project folder in the Finder
  * Click "Delete Project"
  * Click "Download Project" again
- Open `/artisan-test/test-one/aaron` in the S3 staging bucket
- Delete `project-five` in the bucket and its contents.
- Click "Backup"
- In `project-five` locally
- Add a `ai2html-output` folder with an `index.html` file each illustration folder (there should be two)
- Click "Publish Project"
- In the staging bucket, there show now be a `test-one/` and `test-two/` folder in `project-five` with the index file in each.
